x-pack/plugins/task_manager/server/metrics/task_metrics_collector.ts

```diff
@@ -78,7 +99,11 @@ export class TaskManagerMetricsCollector implements ITaskEventEmitter<TaskLifecy
             filter: [
               {
                 bool: {
-                  should: [IdleTaskWithExpiredRunAt, RunningOrClaimingTaskWithExpiredRetryAt],
+                  should: [
+                    IdleTaskWithExpiredRunAt,
+                    RunningOrClaimingTaskWithExpiredRetryAt,
+                    OneOfTaskTypes('task.taskType', searchedTypes),
+                  ],
```

Comment:
To make the filter apply to all tasks.

```suggestion
                  must: [OneOfTaskTypes('task.taskType', searchedTypes)],
                  should: [IdleTaskWithExpiredRunAt, RunningOrClaimingTaskWithExpiredRetryAt],
                  minimum_should_match: 1,
```